pages/docs/workspaces-in-o3/registering-workspaces.mdx

```diff
@@ -0,0 +1,37 @@
+# Registering workspaces
+
+## Workspace Properties
+
+1. **name**: string - Unique identifier for the workspace
+2. **title**: string - Title shown in the workspace header
+3. **load**: Function - A lifecycle function that loads the workspace component (see examples)
+4. **type**: string - type of the workspace (Default = form)
+5. **canHide**: boolean - Can the workspace be collapsed/hidden (Default = false)
+6. **canMaximize**: boolean - Can the workspace be maximized to the full screen (Default = false)
+7. **width**: 'narrow' | 'wider' - Whether the workspace should have a width of 420px (narrow) or 520px (wider; especially for forms with navigation)
+8. **preferredWindowSize**: 'normal' | 'maximized' - Whether to open the workspace in normal view or in maximized view (Default = 'normal')
+
+> 💡 Please note that `canHide` property should be `true` for workspaces with a `type` that corresponds to a siderail nav button.
+
+Registering a new workspace can be simply done in either of the 2 ways:
+
+## Registering workspace in `index.ts` (Recommended)
+
+A new workspace can be registered in the `index.ts` file, using the `registerWorkspace` function from the `@openmrs/esm-patient-common-lib`
+
+```ts copy
+// index.ts
+import { registerWorkspace } from '@openmrs/esm-patient-common-lib';
+
+registerWorkspace({
+  name: string,
+  title: string,
+  load: getAsyncLifecycle(() => import(<path-to-component>), options),
+  type: string,
+  canHide?: boolean = false,
+  canMaximize?: boolean = false,
+  width?: 'narrow' | 'wider' = 'narrow',
+  preferredWindowSize?: 'normal' | 'maximized' = 'normal'
+});
+
+```
```

Comment:
```suggestion

```

You will have to create the file `./gastrodynamics.workspace.tsx` with your workspace component as a default export.
```